### PR TITLE
[FLINK-3523] [Storm-Compatibility] Added SplitStreamMapper to program…

### PR DESCRIPTION
**Pull request by mjsax**
… to get rid of SplitStreamType wrapper

### CODE REVIEW

**Comment by mxm** [comment]
LGTM. Perhaps we could add a test case for this example?

**Comment by mjsax** [comment]
This is an additional example that runs forever -- it was never designed for getting tested. Furthermore, there are already two test for split-stream functionality.

**Comment by mxm** [comment]
IMHO we should have a way to check that examples are working properly. We don't have to address this in this PR though.

**Comment by mjsax** [comment]
Change example slightly (make it run finite and added a check for even/odd stream) and added IT-Case.

**Comment by StephanEwen** [comment]
For the test, two comments:

  - Can you add it to another ITCase (saves bringing an extra mini cluster)
  - Also, it will probably not run on windows. Writing to a temp file and deleting that should make this compatible.

**Comment by mjsax** [comment]
Done. I needed to restructure class hierarchy, because `StreamingMultipleProgramsTestBase` did not provided methods for tmp-file handling...

File: flink-contrib/flink-storm-examples/src/main/java/org/apache/flink/storm/split/SpoutSplitExample.java [hunk 1]
```diff
@@ -84,16 +86,27 @@ public static void main(final String[] args) throws Exception {
 	/**
 	 * Same as {@link VerifyAndEnrichBolt}.
 	 */
-	private final static class Enrich implements MapFunction<Integer, Tuple2<String, Integer>> {
+	public final static class Enrich implements MapFunction<Integer, Tuple2<String, Integer>> {
 		private static final long serialVersionUID = 5213888269197438892L;
 		private final Tuple2<String, Integer> out;
+		private final boolean evenOrOdd; // true: even -- false: odd
```
Comment (mxm) [inline flink-contrib/flink-storm-examples/src/main/java/org/apache/flink/storm/split/SpoutSplitExample.java]: How about `isEven` as name here?

**Comment by mjsax** [comment]
Test fail on Java8 module... Can I merge this @StephanEwen ?

File: flink-contrib/flink-storm-examples/src/main/java/org/apache/flink/storm/split/SpoutSplitExample.java (hunk 1 above)
Comment (mjsax) [inline flink-contrib/flink-storm-examples/src/main/java/org/apache/flink/storm/split/SpoutSplitExample.java]: Sure. Can change it.

**Comment by mxm** [comment]
+1 to merge

**Pull request closed without merging**